vllm/model_executor/model_loader.py

```diff
@@ -44,6 +46,14 @@ def _get_model_architecture(config: PretrainedConfig) -> Type[nn.Module]:
     architectures = getattr(config, "architectures", [])
     for arch in architectures:
         if arch in _MODEL_REGISTRY:
+            # baichuan 2 has different vocab size
+            if ("baichuan" in arch.lower()) and (getattr(config, "vocab_size")
+                                                 == 125696):
+                # baichuan 2 7b and 13b have different intermediate size
+                if getattr(config, "intermediate_size") == 11008:
+                    return BaiChuan2ForCausalLM
+                elif getattr(config, "intermediate_size") == 13696:
+                    return Baichuan2ForCausalLM
```

Comment:
```
                if getattr(config, "intermediate_size") == 11008:
                    arch = "BaiChuan2ForCausalLM"
                elif getattr(config, "intermediate_size") == 13696:
                    arch = "Baichuan2ForCausalLM"
```
Is this better? Since you have already added it in `_MODEL_REGISTRY`.